app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -113,8 +112,10 @@ def index_relation
     end
 
     def find_product_content
-      @product_content = @product.product_content_by_id(params[:id])
+      @product_content = ProductContent.find_by(:id => params[:id])
```

Comment:
So, there is an issue here. When we add your #7629 fix into this code, everything works until you try to enable and disable repositories. This code that I worked on works well in the UI, but is broken when it comes to api requests outside of the UI because an organization would have to be provided and that's a big change to the api. I think this is a good place to start though:

```
def find_product_content
      organization = find_organization
      content = Content.find_by(:cp_content_id => params[:id], :organization_id => organization[:id])
      @product_content = ProductContent.find_by(:id => content[:id])
      fail HttpErrors::NotFound, _("Couldn't find repository set with id '%s'.") % params[:id] if @product_content.nil?
      @product = @product_content.product
      @organization = @product_content.product.organization
    end
```
The issue I run into here is that when content is in 2 different organizations, it shows up twice with Content.where(...). Content.find_by(...) only returns 1 instance. We need the logic in here to return both and store each one in separate products.

At least, this is how I see it. Pinging @jturel because he's been involved with this review. I think we'll need to decide how we want to present this in the api when we have multiple of content in the db.

Reply:
Tagging on to what @johnsonm325 has said: I believe the :id param in the url *is* the cp_content_id. That's from before this API was using Content info cached in the Katello db and going straight to candlepin. Like Michael has mentioned, you can have the same cp_content_id referenced by multiple Katello::Contents across orgs:

```
irb(main):004:0> Katello::Content.where(cp_content_id: 1354)
+-------+---------------+--------------+-----------------------------------+-----------------------------------+---------+-----------------------------------+----------------------------------+-----------------+
| id    | cp_content_id | content_type | name                              | label                             | vendor  | gpg_url                           | content_url                      | organization_id |
+-------+---------------+--------------+-----------------------------------+-----------------------------------+---------+-----------------------------------+----------------------------------+-----------------+
| 7206  | 1354          | yum          | Red Hat Enterprise Linux Load ... | rhel-lb-for-rhel-6-server-eus-... | Red Hat | file:///etc/pki/rpm-gpg/RPM-GP... | /content/eus/rhel/server/6/$r... | 1               |
| 13916 | 1354          | yum          | Red Hat Enterprise Linux Load ... | rhel-lb-for-rhel-6-server-eus-... | Red Hat | file:///etc/pki/rpm-gpg/RPM-GP... | /content/eus/rhel/server/6/$r... | 3               |
+-------+---------------+--------------+-----------------------------------+-----------------------------------+---------+-----------------------------------+----------------------------------+-----------------+
```

The changes proposed to the API are good (this PR and #7629). I don't think we should use cp_content_id as the identifier any more *but* these are breaking API changes and to me at least it feels like this would be the v3 incarnation of the API.

Reply:
@jsherrill any opinions on this PR in terms of how it affects invoking this API?

Reply:
@jlsherrill pinging you again so this PR doesn't go stale.

@sean797 If you have any changes, feel free to push them up in the meantime.